libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -4218,13 +4218,7 @@ string YulUtilFunctions::extractReturndataFunction()
 						data := <emptyArray>()
 					}
 					default {
-						// allocate some memory into data of size returndatasize() + PADDING
-						data := <allocate>(<roundUp>(add(returndatasize(), 0x20)))
```

Comment:
For the record, the main change here is that `allocateMemoryArrayFunction` uses `arrayAllocationSizeFunction` instead of `roundUpFunction` to determine the size.

Reply:
good catch!

Reply:
Fixed in https://github.com/ethereum/solidity/pull/10775